### PR TITLE
first draft of advanced example notebook

### LINKED ISSUE #245: Example - Advanced Topics - Example how to develop your own model
- [ ] How to use the different building blocks?
- [ ] How to combine them to create own new models?
- [ ] E.g. recreate NCF/DLRM or a crazy model

### PR DESCRIPTION
**Pull request by sararb**
Fixes #245 

### Goals :soccer:
- Create the notebook showcasing how to build the DLRM model from scratch

### CODE REVIEW

File: examples/Advanced-Building-your-own-architecture.ipynb [hunk 1]
```diff
@@ -0,0 +1,998 @@
+{
```
Comment (bschifferer) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>I would rename it to something like "Advanced: Define your own architecture"</p><p> </p><p>We can write that we explain it at the example of DLRM</p><p>We do not want to use build in the context of training - building recommender systems includes the 4 stages.</p><p> </p><p>Same for learning objective - Define a model architecture from scratch</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (bschifferer) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>I think that is out of scope for this release.</p><p> </p><p>Why do these Blocks have the postfix <code>Block</code> and the other ones do not? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (bschifferer) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>Why do we call them <code>Blocks</code> and have the postfix <code>Task</code>  ? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (bschifferer) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>is this correct <code>iterated introduced ?</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (bschifferer) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>Can we explicit explain the <code>.connect</code> function? Which blocks do support them? All? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (rnyak) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>sugg:  make models capitalized <code>Merlin Models</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (rnyak) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>May be link to <code>explore-different-models</code> example notebook (https://github.com/NVIDIA-Merlin/models/blob/main/examples/Exploring-different-models.ipynb)  and mention `if you want to check the high-level implementation of DLRM, please refer to .. notebook'. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (rnyak) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>Is it <code>Feature Blocks</code>  or <code>Features Blocks</code> ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (rnyak) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>May be <code>Prediction Tasks</code> or <code>Prediction Heads</code> ?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (rnyak) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p><code>Merlin Models</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (rnyak) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>there is a typo in <code>continous.</code>  should be <code>continuous</code> . I do that typo often too :) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (rnyak) [inline examples/Advanced-Building-your-own-architecture.ipynb:1]: <p>sugg: <code>built-in</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>I see, thanks for the clarification! I updated all the texts using <code>build</code>  and renamed the notebook as well!</p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>As we discussed, I replaced this section by <code>Connects methods</code> listing the methods of the Block class that implement the abstraction to these different combinators.  Regarding your question, I think we chose to add the postfix <code>Block</code> as these classes are applied to a group of <code>Blocks</code>.   </p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>I agree, <code>Task</code> is a better name to reference these classes. I modified the section name accordingly! </p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>No, I forgot to delete the word <code>iterated</code> :D  </p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>It's a method supported by any Block. I added a section <code>Connects Methods</code> that explain this method.  </p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>Thanks for the sugg! I updated all the references. </p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>I added a link to the notebook and the DLRM architecture in the <code>Overview</code> section.  </p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>I used the same naming as in the repo folder (which is <code>Features</code>, and I think it's used to showcase the support of different types of features). Does <code>Feature Blocks</code> sound better for you? </p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>Agree! <code>Prediction Tasks</code> is better.</p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>Done</p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>Same :D thanks for the catch!</p>

Comment (sararb) [reply examples/Advanced-Building-your-own-architecture.ipynb]: <p>Done</p>

**Review by bschifferer — APPROVED** [review]
